x-pack/plugins/files/common/types.ts

```diff
@@ -137,6 +135,25 @@ export type BaseFileMetadata = {
   Compression?: FileCompression;
 };
 
+/**
+ * Metadata add by the files service.
+ */
+export interface FileSystemMetadata {
+  /**
+   * The human-friendly user name of the owner of the file
+   *
+   * @note this field cannot be used to uniquely ID a user. See {@link FileSystemMetadata['uid']}.
+   */
+  owner?: string;
```

Comment:
Can the username of a user change? If yes, should we not store it and instead resolve the name from uid as we need?

Reply:
That's a good point, but I believe this cannot be changed at the moment:

<img width="434" alt="Screenshot 2022-10-31 at 12 57 51" src="https://user-images.githubusercontent.com/8155004/199002634-ebd328c4-1104-498a-a0cf-3cea4db61a85.png">

Reply:
Also, that was a totally random user name! Please don't read into it 😅

Reply:
Sounds good. 

Note: while I was testing, seems like for cloud admin username is not something user friendly but some cloud user id instead, like `"username": "41439991335"`

Reply:
> for cloud admin username is not something user friendly

Right, I believe this is just for the cloud admin user? I would expect the `username` we get from security to be the value that is surfaced elsewhere in the UI, also what is used to generate the default, placeholder avatar.

Reply:
> Right, I believe this is just for the cloud admin user? I would expect the username we get from security to be the value that is surfaced elsewhere in the UI, also what is used to generate the default, placeholder avatar.

🛑  Cloud usernames are always opaque today, and their recent RBAC enhancements allow for multiple people to SSO into deployments, not just the deployment owner.
The display name that we surface in the UI depends on the information available to us, and is NOT unique to the user:

https://github.com/elastic/kibana/blob/f6762956e710c0020c5a90dd6f735caa9add0844/x-pack/plugins/security/common/model/user.ts#L49-L55

Reply:
Thanks for that explanation @legrego !

> Cloud usernames are always opaque today

By "opaque" I am guessing you mean not human readable? And by extension, also not suitable for displaying in the UI more generally except as a final fallback?

If we wanted to provide a human-friendly identifier for a user is there a safe value for us to store? Or should we always resolve this on demand from the user profile data?

Reply:
> By "opaque" I am guessing you mean not human readable? And by extension, also not suitable for displaying in the UI more generally except as a final fallback?

Correct, thanks for clarifying this.

> If we wanted to provide a human-friendly identifier for a user is there a safe value for us to store? Or should we always resolve this on demand from the user profile data?

My recommendation is to resolve this on-demand from the user profile data, if possible. If not, then caching the display name would be _ok_, but keep in mind that the display name:
1) It can go stale. Display names and email addresses can and do change over time
2) Isn't sufficient for uniquely identifying users. I know you know this already -- if it's only stored for informational purposes, then that's more tolerable, but it's not acceptable to use it for access control or auditing.